ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java

```diff
@@ -115,36 +169,83 @@ private void showWelcomeMessage() {
             "Ignoring statements: %s"
-            + "%nOnly CREATE statements can run in standalone mode.",
-            queryMetadata.getStatementString()
+                + "%nOnly DDL (CREATE STREAM/TABLE, DROP STREAM/TABLE, SET, UNSET) "
+                + "and DML(CSAS, CTAS and INSERT INTO) statements can run in standalone mode.",
+            statementPair.getLeft()
         );
         System.err.println(message);
         log.warn(message);
       }
     }
+
+  }
+
+
+  private void handlePersistentQuery(
+      final Statement statement,
+      final String statementString,
+      final Map<String, Object> configProperties) {
+    final Query query;
+    if (statement instanceof CreateAsSelect) {
+      query = ((CreateAsSelect) statement).getQuery();
+    } else if (statement instanceof InsertInto) {
+      query = ((InsertInto) statement).getQuery();
+    } else {
+      throw new KsqlException("Only CSAS/CTAS and INSERT INTO are persistent queries: "
+          + statementString);
+    }
+
+    final QueryMetadata queryMetadata =
+        ksqlEngine.getQueryExecutionPlan(query, ksqlConfig);
+    if (statement instanceof CreateAsSelect) {
+      validateCsasCtas(queryMetadata, (CreateAsSelect) statement);
+    }
+    final List<QueryMetadata> queryMetadataList =
+        ksqlEngine.buildMultipleQueries(statementString, ksqlConfig, configProperties);
+    if (queryMetadataList.isEmpty()
+        || !(queryMetadataList.get(0) instanceof PersistentQueryMetadata)) {
+      throw new KsqlException("Could not build the query: " + statementString);
+    }
+    queryMetadataList.get(0).start();
+  }
+
+  private void validateCsasCtas(final QueryMetadata queryMetadata,
+      final CreateAsSelect createAsSelect) {
+    if (createAsSelect instanceof CreateStreamAsSelect
+        && queryMetadata.getDataSourceType() == DataSource.DataSourceType.KTABLE) {
+      throw new KsqlException("Invalid result type. Your SELECT query produces a TABLE. "
+          + "Please use CREATE TABLE AS SELECT statement instead.");
+    } else if (createAsSelect instanceof CreateTableAsSelect
+        && queryMetadata.getDataSourceType() == DataSourceType.KSTREAM) {
+      throw new KsqlException("Invalid result type. Your SELECT query produces a STREAM. "
+          + "Please use CREATE STREAM AS SELECT statement instead.");
+    }
   }
 
   private static String readQueriesFile(final String queryFilePath) {
-    final StringBuilder sb = new StringBuilder();
-    try (final BufferedReader br = new BufferedReader(new InputStreamReader(
-        new FileInputStream(queryFilePath), StandardCharsets.UTF_8))) {
-      String line = br.readLine();
-      while (line != null) {
-        sb.append(line);
-        sb.append(System.lineSeparator());
-        line = br.readLine();
-      }
+    try {
+      return new String(java.nio.file.Files.readAllBytes(
+          Paths.get("src/test/resources/SampleMultilineStatements.sql")), "UTF-8");
```

Comment:
This was added just for testing? We should remove it or there is only ever going to be one queries file you can use.

Reply:
Good catch, I forgot to remove it. Fixed it.